verify_test.go

```diff
@@ -30,3 +30,56 @@ func TestVerifyNonce(t *testing.T) {
 		t.Errorf("verifyNonce failed unexpectedly: %v", err)
 	}
 }
+
+func TestVerifySignedFields(t *testing.T) {
+	v := url.Values{}
+
+	// Everything properly signed
+	v.Set("openid.signed", "signed,op_endpoint,claimed_id,identity,return_to,response_nonce,assoc_handle")
+	if err := verifySignedFields(v); err != nil {
+		t.Errorf("verifySignedFields failed unexpectedly: %v", err)
+	}
+
+	// No claimed_id/identity
+	v.Set("openid.signed", "signed,op_endpoint,return_to,response_nonce,assoc_handle")
+	if err := verifySignedFields(v); err != nil {
+		t.Errorf("verifySignedFields failed unexpectedly: %v", err)
+	}
+
+	// With claimed_id/identity, but not signed
+	v.Set("openid.claimed_id", "foo")
+	v.Set("openid.identity", "foo")
```

Comment:
I think it's risky that those tests reuse the same url.Values{} instance. You override stuff as you go, but for a test to be complete, one have to read all the above ones as well. Table-based tests would solve that, at the cost of having to repeat some key/values.